### PR TITLE
Fix: Set XHR headers in XHRConnection ctor

### PR DESCRIPTION
**Pull request by georgiosd**
[Review on Reviewable](https://reviewable.io/reviews/angular/angular/2588)

### CODE REVIEW

**Comment by georgiosd** [comment]
I signed it!

File: modules/angular2/src/http/backends/xhr_backend.ts [hunk 1]
```diff
@@ -35,6 +35,13 @@ export class XHRConnection implements Connection {
     this._xhr = new NativeConstruct();
     // TODO(jeffbcross): implement error listening/propagation
     this._xhr.open(RequestMethods[req.method], req.url);
+    var entriesIterator = (req.headers || {}).keys();
```
Comment (vicb) [inline modules/angular2/src/http/backends/xhr_backend.ts]: This would not work in Dart.

Please use `MapWrapper` methods

**Comment by vicb** [comment]
Please also add a unit test

**Comment by georgiosd** [comment]
@vicb I don't really know much about Dart. Can you please be specific about what wouldn't work in Dart? I'll try to remedy it.

**Comment by vicb** [comment]
[Map](https://api.dartlang.org/apidocs/channels/stable/dartdoc-viewer/dart:core.Map) has a different API.

You should rewrite your code using [MapWrapper](https://github.com/angular/angular/blob/df6acedd25b690930148c7c3bae370f2def2feaa/modules/angular2/src/facade/collection.ts#L57) methods.

If you need more info, try to grep the source code.

**Comment by vicb** [comment]
Basically `MapWrapper` is an abstraction layer on top of Dart & JS.

Note than iterators also couldn't be use, `MapWrapper.forEach()` should be ok.

**Comment by georgiosd** [comment]
@vicb ok I wasn't sure if you were pointing to user of the methods or an operator or something else. I'll check it out. Do the other backends need fixing too?

**Comment by vicb** [comment]
I have no idea.

**Comment by georgiosd** [comment]
@vicb Unfortunately I've been hitting roadblocks constantly with 2.0 so I can't spend much more time on this right now as there as pressing deadlines. This was a small internal project so I was hoping ng2 would be good enough to use for that and I could learn/help at the same time. I jumped the gun on that one. I'll be reverting to 1.4 and coming back when a late beta or RC is out!

Happy to open an issue for this so that you can at least track that it's fixed.

**Comment by tbosch** [comment]
Closing this for now.

**Pull request closed without merging**